soc/arm/nordic_nrf/nrf52/dts_fixup.h

```diff
@@ -14,7 +14,7 @@
 
 #define DT_UART_1_NAME			DT_NORDIC_NRF_UARTE_UART_1_LABEL
 
-#define DT_FLASH_DEV_NAME		DT_INST_0_NORDIC_NRF52_FLASH_CONTROLLER_LABEL
+#define DT_FLASH_DEV_NAME		DT_LABEL(DT_INST(0, nordic_nrf52_flash_controller))
```

Comment:
Similar changes need be done in `soc/arm/nordic_nrf/nrf53/dts_fixup.h` and `soc/arm/nordic_nrf/nrf91/dts_fixup.h`. I think it would be good to do it in the same commit that does it for nrf51 and nrf52.